tools/pgo_bolt/train_pgo.py

```diff
@@ -161,12 +166,17 @@ async def profile(args: argparse.Namespace, tmpdir: str, redpanda_bin: str):
 
     finally:
         if omb_proc:
-            await terminate(omb_proc, "omb")
+            omb_status = await terminate(omb_proc, "omb")
+            failed = failed or omb_status != 0
         if cluster_proc:
-            await terminate(cluster_proc, "cluster")
+            cluster_status = await terminate(cluster_proc, "cluster")
+            failed = failed or cluster_status != 0
         if cluster_task:
             await cluster_task
 
+    if failed:
```

Comment:
Maybe you are going to want a log here/before about what failed and with what exit code?